### PR TITLE
Add crashpad

### LINKED ISSUE #55: Persist backtrace on SEH exception
Backtrace shall persist when a debugger is attached after SEH exception was thrown

### PR DESCRIPTION
**Pull request by espkk**
Just add crashpad finally to get rid of dancing with exceptions.
The only stack trace is uploaded for now. To be tweaked in the future.

### CODE REVIEW

**Comment by espkk** [comment]
@np-c0mp1ete could you check what happened to clang build?

**Comment by np-c0mp1ete** [comment]
> @np-c0mp1ete could you check what happened to clang build?

I'm not sure bincrafters support clang builds on Windows. https://github.com/bincrafters/community/pull/1381. Check the list of tests they run on their recipes.

**Comment by espkk** [comment]
> > @np-c0mp1ete could you check what happened to clang build?
> 
> I'm not sure bincrafters support clang builds on Windows. https://github.com/bincrafters/community/pull/1381. Check the list of tests they run on their recipes.

Well, then do we really want to have clang builds on windows?

**Comment by np-c0mp1ete** [comment]
> > > @np-c0mp1ete could you check what happened to clang build?
> > 
> > 
> > I'm not sure bincrafters support clang builds on Windows. [bincrafters/community#1381](https://github.com/bincrafters/community/pull/1381). Check the list of tests they run on their recipes.
> 
> Well, then do we really want to have clang builds on windows?

I'd like to have clang builds in general. It seems there is not much support for clang on windows though. maybe we should indeed drop clang until we are able to compile our code on linux.

**Comment by espkk** [comment]
> > > > @np-c0mp1ete could you check what happened to clang build?
> > > 
> > > 
> > > I'm not sure bincrafters support clang builds on Windows. [bincrafters/community#1381](https://github.com/bincrafters/community/pull/1381). Check the list of tests they run on their recipes.
> > 
> > Well, then do we really want to have clang builds on windows?
> 
> I'd like to have clang builds in general. It seems there is not much support for clang on windows though. maybe we should indeed drop clang until we are able to compile our code on linux.

Maybe there is a way to add msvc-clang build instead? It is literally clang that is much more supported on windows I guess
Anyway, I'm afraid that the current configuration will prevent us from integrating some cool things
We probably should discuss that

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,10 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
 # Setup third-party dependencies
 include(conan)
+conan_add_remote(NAME bincrafters
+    URL https://api.bintray.com/conan/bincrafters/public-conan
```
Comment (np-c0mp1ete) [inline CMakeLists.txt]: that's a deprecated url. check the [docs](https://bincrafters.readthedocs.io/en/latest/using_packages.html#adding-the-bincrafters-repository-as-a-conan-remote) and how @q4a [integrated sdl2](https://github.com/q4a/storm-engine/commit/0a78c174129c2b265324e5b155766b80bb3076cd#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR10)

File: src/apps/ENGINE/crashpad.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#include "vfile_service.h"
+
+#include <client/crash_report_database.h>
+#include <client/crashpad_client.h>
+#include <client/settings.h>
+
+namespace storm::except
+{
+
+bool startCrashHandler()
```
Comment (np-c0mp1ete) [inline src/apps/ENGINE/crashpad.cpp]: Most of the codebase uses CamelCase style. Probably we should stick to it.

**Review by np-c0mp1ete — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (espkk) [reply CMakeLists.txt]: Not on binary? :O

File: src/apps/ENGINE/crashpad.cpp (hunk 2 above)
Comment (espkk) [reply src/apps/ENGINE/crashpad.cpp]: Our coding style describes exactly this naming...

Comment (np-c0mp1ete) [reply src/apps/ENGINE/crashpad.cpp]: I know, I left a comment there in this regard.

File: CMakeLists.txt (hunk 1 above)
Comment (np-c0mp1ete) [reply CMakeLists.txt]: what do you mean?

Comment (espkk) [reply CMakeLists.txt]: T9 :( I meant bintray. I believe it's not deprecated but rather a release repo link...

Comment (espkk) [reply CMakeLists.txt]: Ok I see it's getting closed soon

File: src/apps/ENGINE/crashpad.cpp (hunk 2 above)
Comment (espkk) [reply src/apps/ENGINE/crashpad.cpp]: as discussed, let's leave it as is right now as global reformat incoming

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -7,6 +7,10 @@ if (STORM_ENABLE_STEAM)
 endif()
 
 if (MSVC)
+    # Output to the same directory regardless of [CONFIG]
+    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_DEBUG "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
```
Comment (np-c0mp1ete) [inline src/CMakeLists.txt]: I tested it on my PC and I don't have this issue with subdirectories inside `bin`.

**Review by np-c0mp1ete — COMMENTED** [review]
Thanks for the PR! How can I access the sentry dashboard?

File: src/CMakeLists.txt (hunk 3 above)
Comment (espkk) [reply src/CMakeLists.txt]: as discussed, this problem appears when generating sln rather than using CMakeLists directly

**Review by np-c0mp1ete — APPROVED** [review]

**Pull request merged**